mobile/evidence/ios/Features/Sources/Profile/StatusViewModel.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  StatusViewModel.swift
+//  Evidence
+//
+//  Created by Cris Messias on 09/11/23.
+//
+
+import SwiftUI
+import Models
+
+public class StatusViewModel: ObservableObject {
+    @Published private(set) var currentState: Profile = .active
+    @Published private(set) var showModal = false
+    @Published var statusInput: String = ""
+    @Published private(set) var showAlert = false
+    @Published private(set) var isButtonSaveTapped = false
+    @Published private(set) var isClearButtonShowing = false
+    
+    public init(currentState: Profile = .active, showModal: Bool = false, statusInput: String = "", showAlert: Bool = false, isButtonSaveTapped: Bool = false, isClearButtonShowing: Bool = false) {
+        self.currentState = currentState
+        self.showModal = showModal
+        self.statusInput = statusInput
+        self.showAlert = showAlert
+        self.isButtonSaveTapped = isButtonSaveTapped
+        self.isClearButtonShowing = isClearButtonShowing
+    }
+    
+    func isModalShowing() { 
+        showModal.toggle()
+    }
+    
+    func saveStatusButton() {
+        if statusInput.isEmpty {
+            showModal = true
+        } else {
+            showModal = false
+        }
+    }
+    
+    func toggleButton() {
+        isClearButtonShowing.toggle()
+    }
+    
+    func clearStatus() {
+        statusInput = ""
+    }
+    
+    func isAlertShowing() {
+        showAlert = true
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1) {
+            self.showAlert = false
+        }
```

Comment:
Aqui estamos criando um retain cycle que vai causar um problema de memória no nosso app. Depois leia sobre isso aqui: https://docs.swift.org/swift-book/documentation/the-swift-programming-language/automaticreferencecounting/#How-ARC-Works
```suggestion
        DispatchQueue.main.asyncAfter(deadline: .now() + 1) { [weak self] in
            self?.showAlert = false
        }
```

Reply:
Eu comecei a ler sobre [weak self] mas ainda não sei bem como usar. Vou ler o artigo :D